internal/telemetry/client.go

```diff
@@ -443,6 +459,10 @@ func (c *client) flush() {
 				log("submission error: %s", err.Error())
 			}
 		}
+
+		if callback != nil {
+			callback()
+		}
 	}()
```

Comment:
The whole code that is ran async is this one:
```go
        go func() {
		for _, r := range submissions {
			err := r.submit()
			if err != nil {
				log("submission error: %s", err.Error())
			}
		}

		if callback != nil {
			callback()
		}
        }()
```

How about passing a simple boolean to do something like this?
```go
        submit := func() {
		for _, r := range submissions {
			err := r.submit()
			if err != nil {
				log("submission error: %s", err.Error())
			}
		}

		if callback != nil {
			callback()
		}
        }
        if stopSync {
		submit()
        } else {
		go submit()
        }
```

Reply:
yes, make sense, I did the change in [9c8f9e7](https://github.com/DataDog/dd-trace-go/pull/2963/commits/9c8f9e7ce3754968b03f8fa750de3b137371a585)